salt/utils/dictupdate.py

```diff
@@ -50,7 +54,7 @@ def update(dest, upd, recursive_update=True, merge_lists=False):
             elif isinstance(dest_subkey, list) \
                      and isinstance(val, list):
                 if merge_lists:
-                    dest[key] = dest.get(key, []) + val
+                    dest[key] = sorted((set(dest_subkey + val)))
```

Comment:
And, heh, to me the set is far simpler to comprehend than the list comprehension, so I'm gonna quibble about the "simple" characterization. 😜